### PR TITLE
Added additional models to conftest

### PR DESCRIPTION
**Pull request by MacMcIrish**

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -1,28 +1,103 @@
...
+            'school_id',
+            ForeignKey(School.id, ondelete='cascade'),
+            primary_key=True
+        ),
+        Column(
+            'classroom_id',
+            ForeignKey(Classroom.id, ondelete='cascade'),
+            primary_key=True
+        )
+    )
```
Comment (simlu) [inline tests/conftest.py]: I think cascade only makes sense in one direction here

File: run_tests.py [hunk 2]
```diff
@@ -1,15 +1,18 @@
...
         '--cov-report=html',
         '--cov-report=term-missing:skip-covered',
-        '--cov-config=.coveragerc',
-        '-s'
-    ])
+        '--cov-config=.coveragerc'
+    ]
+    if '--verbose' in sys.argv:
+        args.append('-s')
+    pytest.main(args)
```
Comment (simlu) [inline run_tests.py]: Needs to be detailed in tests (plus how to run tests)

File: tests/conftest.py [hunk 3]
```diff
@@ -1,28 +1,103 @@
...
+            'teacher_id',
+            ForeignKey(Teacher.id, ondelete='cascade'),
+            primary_key=True
+        ),
+        Column(
+            'student_id',
+            ForeignKey(Student.id, ondelete='cascade'),
+            primary_key=True
+        )
+    )
```
Comment (simlu) [inline tests/conftest.py]: Not sure the cascade is appropriate here

Comment (MacMcIrish) [reply tests/conftest.py]: I've removed it, I misread what was in our generate_linkage_table function

File: README.md [hunk 4]
```diff
@@ -24,7 +24,13 @@
...
     ~ Where are the tests?
 
-      ....
+      Start the tests with:
+      
+      $ python run_tests.py [options]
+      
+      Options:
+        --verbose
+        Output all debug information (warning: interferes with test progress output)
```
Comment (simlu) [inline README.md]: remove word `all`

File: tests/test_ModelFilter.py [hunk 5]
```diff
@@ -0,0 +1,28 @@
+
+
+class TestModelFilter(object):
+
+    def test_filter_by_id(self, Teacher):
+        teacher_id = Teacher().save().id
+        assert Teacher.filter({'id': teacher_id}).one().id == teacher_id
+
+    def test_filter_by_one_to_many_relationship(self, Teacher, Student):
```
Comment (simlu) [inline tests/test_ModelFilter.py]: ?????

File: tests/test_ModelFilter.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        student = Student(name='foo')
```
Comment (simlu) [inline tests/test_ModelFilter.py]: lets start using faker

Comment (simlu) [reply tests/test_ModelFilter.py]: Not for this pr though

File: tests/test_ModelFilter.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+        teacher_id = Teacher(students=[student]).save().id
+        assert Teacher.filter({
+            'students.name': student.name
+        }).one().id == teacher_id
+
+    def test_filter_by_one_to_one_relationship(self, Classroom, Teacher):
```
Comment (simlu) [inline tests/test_ModelFilter.py]: ?????????

File: tests/test_ModelFilter.py [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+        teacher = Teacher()
+        classroom_id = Classroom(teacher=teacher).save().id
+        assert Classroom.filter({
+            'teacher.id': teacher.id
+        }).one().id == classroom_id
+
+    def test_filter_by_foreign_key(self, Classroom, Teacher):
```
Comment (simlu) [inline tests/test_ModelFilter.py]: test_filter_by_column?

File: tests/conftest.py [hunk 9]
```diff
@@ -1,28 +1,90 @@
...
+
+    return Classroom
+
+
+@pytest.fixture(scope='session')
+def Teacher(Classroom):
     class Teacher(Model):
         __tablename__ = 'teacher'
+
+        classroom_id = Column(Integer, ForeignKey(Classroom.id))
```
Comment (simlu) [inline tests/conftest.py]: unique

File: tests/test_ModelFilter.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+
+    def test_filter_by_one_to_one_relationship(self, Classroom):
+        assert Classroom.filter({
+            'teacher.id': self.teacher_id
+        }).one().id == self.classroom_id
+
+    def test_filter_by_foreign_key(self, Classroom):
+        assert Classroom.filter({
+            'teacher_id': self.teacher_id
+        }).one().id == self.classroom_id
```
Comment (simlu) [inline tests/test_ModelFilter.py]: add many to one and one to many

**Pull request merged**